sigmastate/src/main/scala/org/ergoplatform/ErgoBoxCandidate.scala

```diff
@@ -72,35 +87,49 @@ class ErgoBoxCandidate(val value: Long,
     }
   }
 
-  override def hashCode(): Int =
+  // TODO HF: fix hashCode, it should be consisten with [[equals]] and use [[bytesWithNoRef]]
+  override def hashCode(): Int = {
     ScalaRunTime._hashCode((value, ergoTree, additionalTokens, additionalRegisters, creationHeight))
+  }
 
   override def toString: Idn = s"ErgoBoxCandidate($value, $ergoTree," +
     s"tokens: (${additionalTokens.map(t => ErgoAlgos.encode(t._1) + ":" + t._2).toArray.mkString(", ")}), " +
     s"$additionalRegisters, creationHeight: $creationHeight)"
 
-  lazy val tokens: Map[ModifierId, Long] = 
+  /** Additional tokens stored in the box. */
+  lazy val tokens: Map[ModifierId, Long] =
     additionalTokens
       .toArray
       .map(t => bytesToId(t._1) -> t._2)
       .toMap
 }
 
 object ErgoBoxCandidate {
+  /** Default value of Reference Register (R3) returned by ErgoBoxCandidate. */
```

Comment:
This comment is not true, as register R3 also contains creating height

Reply:
clarified.